### ISSUE TITLE
Latex to unicode does not work in CoqIDE 8.10+beta3

### ISSUE THREAD

**Issue by julesjacobs** [coq/coq#10842]
#### Description of the problem
Pressing shift+space or invoking the command from the menu only inserts a space and does not convert the Latex, such as \pi, to unicode.

#### Coq Version
8.10+beta3, Ubuntu 19.04.

**Comment by Zimmi48** [comment]
Using the menu also inserts a space? This is puzzling!?
FTR on my machine, the default keyboard shortcut does not work because it is bound already to insertion on a non-breaking space, but the menu does work (with Coq 8.10+beta2).

**Comment by julesjacobs** [comment]
Indeed, the menu also inserts a space...I will try beta2.

**Comment by julesjacobs** [comment]
I do not have this problem on beta2.

**Comment by Zimmi48** [comment]
That's really weird. OTOH I have just tested 8.10+beta3, and master and I still can't reproduce this problem.

**Comment by Zimmi48** [comment]
How did you install Coq? Can you try going back to 8.10+beta3 to check that the problem is still there?

**Comment by julesjacobs** [comment]
I'm using opam. I'm already going back to beta3 :)

**Comment by julesjacobs** [comment]
It now works on beta3. The installation of beta2 possibly leaves behind some files which makes it work on beta3...?

**Comment by Zimmi48** [comment]
When you found the bug on beta3, did you try closing and reopening CoqIDE? Or did you only test it the first time you opened it?

**Comment by julesjacobs** [comment]
I tried closing and reopening.

**Comment by Zimmi48** [comment]
In principle, the only files that are left by CoqIDE are in `./config/coq`. You can have a try at moving them around, and reopening CoqIDE, but I doubt that this will have an effect...

@charguer, the author of this feature, might have a better idea.

**Comment by julesjacobs** [comment]
It still works even after moving that directory. By the way, using the menu does insert a space when there is no Latex command before the cursor. Perhaps CoqIDE was missing the file that contains the list of latex commands?

**Comment by Zimmi48** [comment]
In principle, a file (`./config/coq/coqide.bindings`) is only needed to extend the default bindings, which should work out of the box...

**Comment by julesjacobs** [comment]
Okay, so I just realised that the problem is completely different. My apologies for the misdirection. The problem occurs when there is whitespace behind the cursor. Let _ represent whitespace and | the cursor. Then if we do \pi|_ and press shift+space, it does not work. If the _ is removed or replaced by a letter, it does work.

**Comment by Zimmi48** [comment]
No worries. Indeed, I am able to reproduce this. This bug is probably there since the feature was introduced. Fortunately, it is not as worrying, and thus not as high priority. But this is still a bug.

@charguer Do you want to look into this one?

**Comment by charguer** [comment]
The shift+space is meant to work immediately after typing the prefix, e.g. "\pi".
It has never been meant to work if the cursor follows a space, e.g. "\pi ".
Is that what you're talking about? 
Then, I wouldn't say it's a bug; at best a feature wish :)

If we are moving to VSCode in less than 1 year, we probably shouldn't worry about CoqIDE's new features?

**Comment by julesjacobs** [comment]
Note that the cursor is not after the space, but before it. I did not expect \pi_| to work, but \pi|_.
The issue isn't so much a space as a newline. When typing Lemma f : a \-> the \-> won't convert unless it's the last line in the file.

**Comment by Zimmi48** [comment]
>  Is that what you're talking about?

Nope.

> If we are moving to VSCode in less than 1 year, we probably shouldn't worry about CoqIDE's new features?

What?! Then, let's remove this feature outright for CoqIDE, no?

**Comment by charguer** [comment]
I found the issue, it's an offby one thing. Will try a fix.

**Comment by charguer** [comment]
I'm not sure from which branch I should do a pull request. (please provide instructions if you wish a proper pull request). The patch is simple:

in file `ide/wg_ScriptView.ml`

Replace:
````
     let rec get_line_start iter =
      if iter#starts_line || Glib.Unichar.isspace iter#char then iter
      else get_line_start iter#backward_char
````

with

````
    let rec get_line_start iter =
      if iter#starts_line then iter else
      let prev = iter#backward_char in
      if Glib.Unichar.isspace prev#char then prev
      else get_line_start prev
      in
````

My own tests are successful, but further testing would be desirable of course.

**Comment by julesjacobs** [comment]
Awesome! This feature is a life saver for me. Thanks a lot for implementing it.

**Comment by charguer** [comment]
@julesjacobs : glad you like it. Please test it extensively if you plan to rely on it.

@Zimmi48 : should I just branch from the current 8.10 branch to make the PR?

**Comment by Zimmi48** [comment]
> should I just branch from the current 8.10 branch to make the PR?

No, always from master. I have a look at the contributing guide maybe :wink:

**Comment by charguer** [comment]
See https://github.com/coq/coq/pull/10852
